lib/web_ui/lib/src/engine/semantics/focusable.dart

```diff
@@ -184,9 +177,7 @@ class AccessibilityFocusManager {
 
     EnginePlatformDispatcher.instance.invokeOnSemanticsAction(
       target.semanticsNodeId,
-      acquireFocus
-        ? ui.SemanticsAction.didGainAccessibilityFocus
-        : ui.SemanticsAction.didLoseAccessibilityFocus,
+      ui.SemanticsAction.focus,
```

Comment:
don't we also want to send a11yfocus as well?

Reply:
That would be a wrong thing to do. On the web, accessibility focus is not observable through any platform API. If the accessibility focus is indeed detached from the input focus, sending `didGainAccessibilityFocus` together with `focus` would be incorrect. I think instead, we want the framework to be resilient to the absence of accessibility focus signals (it seems to be the case already for our own widgets; according to https://github.com/flutter/flutter/issues/83809#issuecomment-2127888048 it's only used for highlighting, which can be achieved at least on desktop through input focus).

Reply:
> If the accessibility focus is indeed detached from the input focus, sending didGainAccessibilityFocus together with focus would make the framework believe a wrong widget has accessibility focus.

but on the web they are not detached, right? or at least we can't tell whether something receives only the a11y focus. but when something receives focus, it must also have received a11y focus?

Reply:
They can be detached on the web too. For example, you can uncheck the following checkbox in macOS VoiceOver utility:

![Screenshot 2024-05-31 at 12 44 06 PM](https://github.com/flutter/engine/assets/211513/eb6021be-6e56-4773-b731-11268b417675)

With that checkbox off, you can move the accessibility focus around without moving the input focus. Here's an example of VoiceOver focus being on a text field inside the web page, but when typing your input goes into the address bar instead:

![Screenshot 2024-05-31 at 12 46 29 PM](https://github.com/flutter/engine/assets/211513/afead657-3cbd-4c6c-8df6-6894981d6af1)

Here's another example, where the a11y focus is on the last name field, but I'm entering the first name field:

![Screenshot 2024-05-31 at 12 57 05 PM](https://github.com/flutter/engine/assets/211513/51d6931c-22cd-4903-8935-99e7f7519e91)

You need to take special action to activate the widget under VoiceOver cursor, e.g. Control + Option + Space.

However, there's no listeners we can hook up on the web page to tell where the VoiceOver cursor is headed. So it's not observable to us. That's not a real problem though. The browser already does whatever it needs and determines when input focus should move into the field, and send us a DOM "focus" event. The web engine will simply forward it to the framework as `SemanticsAction.focus`.

I must say, this detached mode feels confusing to me. I think when both kinds of focus are synchronized, it's easier to use the UI. VoiceOver itself seems to be giving incorrect instructions. For example, when you move focus to a text field, it says "To enter text in this field, type", which is incorrect in this mode as the text field does not yet have focus.

Reply:
I see, so the web just not going to use a11y focus. Now I think of it this is probably correct since native web can't tell which dom has the a11y focus either.